src/lib/crypto/kimchi_backend/gadgets/common.ml

```diff
@@ -0,0 +1,148 @@
+(* Common gadget helpers *)
+
+open Core_kernel
+
+(* field_bits_to_field - Create a field element from contiguous bits of another
+ *
+ *   Inputs:
+ *     field_element: source field element
+ *     start:         zero-indexed starting bit offset (or -1 to denote the last bit)
+ *     stop:          zero-indexed stopping bit index (or -1 to denote the last bit)
+ *
+ *   Output:
+ *     New field element created from bits [start, stop) of field_element input,
+ *     placed into the lowest possible bit position, like so
+ *
+ *        start     stop
+ *             \   /
+ *       [......xxx.....] field_element
+ *       [...........xxx] output
+ *       msb          lsb
+ *)
+let field_bits_to_field (type f)
+    (module Circuit : Snarky_backendless.Snark_intf.Run with type field = f)
+    (field_element : Circuit.Field.t) (start : int) (stop : int) =
+  let open Circuit in
+  (* Check range is valid *)
+  if stop <> -1 && stop <= start then
+    invalid_arg "stop offset must be greater than start offset" ;
+
+  (* Create field element *)
+  exists Field.typ ~compute:(fun () ->
+      (* Convert field element to bits (boolean list) *)
+      let bits =
+        Field.Constant.unpack @@ As_prover.read Field.typ field_element
+      in
+      let stop = if stop = -1 then List.length bits - 1 else stop in
+      (* Convert bits range (boolean list) to field element *)
+      Field.Constant.project @@ List.take (List.drop bits start) (stop - start) )
```

Comment:
```suggestion
      Field.Constant.project @@ List.slice bits start stop
```

See https://ocaml.janestreet.com/ocaml-core/109.20.00/doc/core/List.html

Reply:
awesome! ty!

Reply:
List.slice implementation should perform better, right?  It has fewer allocations?